File: codeforces/1600~1699/1673/b.cc
#include <bits/stdc++.h>

using namespace std;

using i64 = long long;

void solve() {
  string s;
  cin >> s;
  int n = s.length();

  string chars = s;
  sort(chars.begin(), chars.end());
  chars.erase(unique(chars.begin(), chars.end()), chars.end());

  for (char ch : chars) {
    vector<int> count(26);
    for (int i = 0; i < n; ++i) {
      if (s[i] == ch) {
        fill(count.begin(), count.end(), 0);
      } else {
        int c = s[i] - 'a';
        if (++count[c] >= 2) {
          cout << "NO" << endl;
          return;
        }
      }
    }
  }

  cout << "YES" << endl;
}

int main(void) {
  ios::sync_with_stdio(false);
  cin.tie(0);

  int T = 0;
  cin >> T;
  for (int t = 0; t < T; ++t) {
    solve();
  }
  return 0;
}

File: codeforces/1600~1699/1673/b.rs
use std::io::{stdin, stdout, BufWriter, Write};
use std::str::FromStr;
use std::fmt::Debug;

fn read_int<T: FromStr>() -> T where <T as FromStr>::Err: Debug {
    let mut s = String::new();
    stdin().read_line(&mut s).unwrap();
    s.trim().parse::<T>().expect("invalid input")
}

fn read_str() -> String {
    let mut s = String::new();
    stdin().read_line(&mut s).unwrap();
    s.trim().to_string()
}

fn solve() {
    let stdout = stdout();
    let mut writer = BufWriter::new(stdout.lock());

    let s = read_str();
    let n = s.len();
    let mut chars: Vec<char> = s.chars().map(|ch| ch).collect();
    chars.sort();
    chars.dedup();

    let b = s.as_bytes();
    for ch in chars {
        let mut count = vec![0; 26];
        for i in 0..n {
            if b[i] == ch as u8 {
                for k in 0..26 {
                    count[k] = 0;
                }
            } else {
                let c = (b[i] - b'a') as usize;
                count[c] += 1;
                if count[c] >= 2 {
                    writeln!(writer, "NO").ok();
                    return;
                }
            }
        }
    }
    writeln!(writer, "YES").ok();
}

fn main() {
    let n = read_int::<i32>();
    for _t in 0..n {
        solve();
    }
}
